README.md

```diff
@@ -146,6 +146,140 @@ Module blocks could be transpiled to either data URLs, or to a module in a separ
 
+## FAQs
+
+### Should we really allow creation of workers using module blocks?
+
+In my opinion: Yes. The requirement that workers are in a separate file is one of the most prominent pieces of feedback about why workers are hard to adopt. That’s why so many resort to Blob URLs or Data URLs, bringing along all kinds of difficulties, especially relating to paths and CSP. The risk here is that people start spawning a lot workers without regards to their cost, but I think the benefits of lowering the barrier to workers as an important performance primitive outweigh the risks.
+
+### Can you close over variables? Can you reference values outside the module block?
+
+No. Just like a separate file containing a ES module, you can only reference the global scope and import other modules.
+
+### Can you _statically_ import other modules?
+
+Yes. Just like with a separate file. This is completely valid:
+
+```js
+const m = module {
+  import myApiWrapper from "./api-wrapper.js";
+
+  await someTopLevelAwaitLogic();
+}
+```
+
+### What about a nice shorthand for modules that just contain one function?
+
+Something like this?
+
+```js
+const m = module (x) => x**2;
+
+// equivalent to
+
+const m = module {
+  export default (x) => x**2;
+}
+```
+
+It is unclear whether this will actually be a common pattern, as it not only requires the module to have exactly one function, but also that it does not need any static imports. For now we plan to leave the MVP without such a shorthand and observe what kind of usage patterns emerge. If import-free single-function modules become a frequently recurring pattern, a module shorthand seems like a good idea to add.
+
+### Can Module Blocks help with bundling?
+
+Mostly no. In _some_ scenarios Module Blocks can be sufficient to bundle:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const { count } = await import(count);
+const { uppercase } = await import(uppercase);
+
+console.log(count()); // 1
+console.log(uppercase("daniel")); // "DANIEL"
+```
+
+In the _general_ case, however, modules need to refer to each other. For that to work Module Blocks would need to be able to close over modules, which they can’t do:
+
+```js
+const count = module {
+  let i = 0;
+
+  export function count() {
+    i++;
+    return i;
+  }
+};
+
+const uppercase = module {
+  export function uppercase(string) {
+    return string.toUpperCase();
+  }
+};
+
+const combined = module {
+  const { count } = await import(count);
+  const { uppercase } = await import(uppercase);
+
+  console.log(count()); // 1
+  console.log(uppercase("daniel")); // "DANIEL"
+};
+
+// ReferenceError as we can't close over count or uppercase!!
+```
+
+To address the bundling problem, Dan Ehrenberg is maintaining a separate [proposal/idea][js module bundles].
```

Comment:
Note for later: I'd like to instead refer to a version of web bundles eventually, but we can change this link once the writeup for that is improved a bit more.